rfcs/0017-long-fork-consensus.md

```diff
@@ -0,0 +1,84 @@
+## Summary
+[summary]: #summary
+
+We currently do not support long forks. This means that to join the network you need a recent checkpoint, in other words to have always been online or to trust another party that has always been online.
+
+This adds support for a client to join the network at any point of time, regardless of if its been online recently. It adds this safely under a [50% + epsilon] security assumption. It also includes a mechanism to restore a chain to strength if this assumption was to be temporarily violated.
+
+## Motivation
+
+[motivation]: #motivation
+
+Its not realistic for clients to have always been online, and it is a poor trust model to require clients to find a trusted third party (high friction and not always feasible).
+
+With this feature, clients will be able to join the network from any device at any time.
+
+## Detailed design
+
+[detailed-design]: #detailed-design
+
+### Stage 1: min-window
+
+This version of the implementation is correct given that [50% + epsilon] of stake has always been participating and honest. 
+
+* Add a new field to protocol_state called "min_window", initally set to 1
+* Add a new field to protocol_state called "current_window", initially set to 0
+* If `FLOOR(previous_protocol_state.slot_number/(8k)) == FLOOR(current_protocol_state.slot_number/(8k))`
+  * set `current_protocol_state.current_window` to `previous_protocol_state.current_window + 1`
+* If `FLOOR(previous_protocol_state.slot_number/(8k)) == FLOOR(current_protocol_state.slot_number/(8k)) + 1`
+  * set "current_protocol_state.min_window" to `MIN(current_protocol_state.current_window, previous_protocol_state.min_window)`
+  * set `current_protocol_state.current_window` to `1`
+* else
+  * set "current_protocol_state.min_window" to `0`
+  * set `current_protocol_state.current_window` to `1`
+
```

Comment:
`min_window` is initialized at 1, and is either set to 0 or to the minimum of the `current_window` and the previous `min_window`. So it can't ever grow.

Reply:
That's as intended. You can imagine if participation was fixed at some portion of the total stake, each window generates a sample from some distribution centered on the participation, and the "min-window" will asymptote over time towards a low percentile of this distribution (but as shown in genesis, never too low that the honest chain can't be distinguished from an adversary's chain).

This however does not account for events outside of this assumption (say, the internet goes down and an epoch is skipped). Part 2 below fixes this and allows for a recovery state (you can imagine a "saw-tooth" pattern for min-window once this is implemented)

Reply:
oh oops realized you're talking about min-length being 1 and current window always incremented - will fix. Hope the above was helpful tho for some of the intuition

Reply:
changed `min-window` to `8k` should be correct now